BOOM/Code/Services/config_service.lua

```diff
@@ -0,0 +1,192 @@
+    end
+
+    -- populate from includes
+    for k, v in pairs(namespace) do
+        if type(v) == "table" then
+            include(namespace, k, v)
+        end
+    end
+
+    clear(includedChecklist)
+end
+
+local function recursiveLoad(path, requirePath)
+    for k in LuaFileSystem.dir(path) do
+        if k ~= "." and k ~= ".." then
+            local fileAttrib = LuaFileSystem.attributes( path .. "/" .. k )
+            if fileAttrib.mode == "directory" then
+                recursiveLoad(path .. "/" .. k, requirePath .. "." .. k)
+            elseif fileAttrib.mode == "file" and string.match(k, "[.lua]$") then
+                -- Make a new string based on the require path, so it doesn't carry over for the next load
+                local finalRP = requirePath .. "." .. k
+                -- Create a substring that removes ".lua" from the end
+                local luaPostfixLength = 4
+                finalRP = string.sub(finalRP, 1, string.len(finalRP) - luaPostfixLength)
+                load(finalRP)
+            end
+        end
+    end
+end
+
+------------------------------------------------------------------------------------
+-- Local Functions - Getter
+------------------------------------------------------------------------------------
+local splitId = {}
+local function findAndLoadAsset(searchSpace, assetKey, loadFunction, result)
+    for k, v in pairs(searchSpace) do
+        if k == assetKey then
+            if v == "table" then
+                for i = 1, #v do
+                    if not result[v[i]] then -- Avoiding duplicate load calls
+                        result[v[i]] = loadFunction(v[i])
+                    end
+                end
+            elseif not result[v] then -- Avoiding duplicate load calls
+                result[v] = loadFunction(v)
+            end
+        elseif type(v) == "table" then
+            findAndLoadAsset(v, assetKey, loadFunction, result)
+        end
+    end
+end
+
+------------------------------------------------------------------------------------
+-- ConfigService Functions
+------------------------------------------------------------------------------------
+
+function ConfigService:fromId(configId)
+    splitId = split(configId, ".", splitId)
+    if #splitId < 2 then
+        error("Invalid Id: " .. configId .. ". Did you forget the namespace?")
+        return nil
+    end
+
+    local namespace = Configs[splitId[1]]
+    if not namespace then
+        error("Invalid Id:" .. configId .. ". Could not find namespace: " .. splitId[1])
+        return nil
+    end
+
+    local config = getConfig(splitId, 2, namespace)
+    clear(splitId)
+    return config
+end
+
+function ConfigService:buildAssetTable(namespaceKey, assetKey, loadFunction, result)
+    local namespace = Configs[namespaceKey]
+    if not namespace then
+        error("Cannot find namespace \"" .. namespaceKey .. "\" in Configs. Cannot build Asset Table.")
+        return result
+    end
+    return find
```

Comment:
undefined variable, is this intentional?

Reply:
Ya that was a bug, it was supposed to be calling the recursive findAndLoadAsset() function..